StripeFinancialConnections/StripeFinancialConnections/Source/Common/HostController.swift

```diff
@@ -96,41 +99,31 @@ extension HostController: HostViewControllerDelegate {
     ) {
         delegate?.hostController(self, didReceiveEvent: FinancialConnectionsEvent(name: .open))
 
-        if synchronizePayload.manifest.isProductInstantDebits {
-            continueWithWebFlow(synchronizePayload.manifest)
-        } else {
-            let flowRouter = FlowRouter(
-                synchronizePayload: synchronizePayload,
-                analyticsClient: analyticsClient
-            )
-            defer {
-                // no matter how we exit this function
-                // log exposure to one of the variants if appropriate.
-                flowRouter.logExposureIfNeeded()
-            }
-
-            guard flowRouter.shouldUseNative else {
-                continueWithWebFlow(synchronizePayload.manifest)
-                return
-            }
-
-            navigationController.configureAppearanceForNative()
-
-            let dataManager = NativeFlowAPIDataManager(
-                manifest: synchronizePayload.manifest,
-                visualUpdate: synchronizePayload.visual,
-                returnURL: returnURL,
-                consentPaneModel: synchronizePayload.text?.consentPane,
-                apiClient: api,
+        let flowRouter = FlowRouter(
+            synchronizePayload: synchronizePayload,
+            analyticsClient: analyticsClient
+        )
+
+        let flow = flowRouter.flow
+        v1AnalyticsClient.log(
+            analytic: FinancialConnectionsSheetFlowDetermined(
                 clientSecret: clientSecret,
-                analyticsClient: analyticsClient
-            )
-            nativeFlowController = NativeFlowController(
-                dataManager: dataManager,
-                navigationController: navigationController
-            )
-            nativeFlowController?.delegate = self
-            nativeFlowController?.startFlow()
+                flow: flow,
+                killswitchActive: flowRouter.killswitchActive
+            ),
+            apiClient: .shared
+        )
+
+        switch flow {
+        case .webInstantDebits:
+            continueWithWebFlow(synchronizePayload.manifest)
+        case .nativeInstantDebits:
+            // Not currently supported.
+            break
```

Comment:
just making a mental note: I believe this is safe and there's no need to add error logging, or route to web flow, because we can verify with code logic that this will never hit anyway